### DISCUSSION TITLE
Groundwater models - boundary conditions

### DISCUSSION THREAD

**Discussion by josebastiase** [idaholab/moose#18031] [Q&A Modules: General] [answered]
Hi,
I'm working on a groundwater model (HM - coupled) to simulate land subsidence due to heavy groundwater extraction. My questions:

1. For the top layer of the model which is unsaturated and drained, I'm using `PorousFlowPiecewiseLinearSink` to fix the porepressure at the boundaries. I would like to match the boundary conditions with the water head level observed in the field. But I guess that is not really possible with `PorousFlowPiecewiseLinearSink`, but trying different values of the `flux_function` flag one can get really close. Is this approach correct/ideal?

2. The model is bounded in one side by a river (I'm using the water level of the river as BC). What kind of boundary condition would be more suitable in this situation? I would use `PorousFlowPiecewiseLinearSink` again but a line sink/source may also do the job?

Thanks in advance!
Jose

**Comment by WilkAndy** [comment, marked as answer]
What you're doing is fine, but below I've written some additional thoughts.

(1) You might like to fix the porepressure (or head) using a `DirichletBC` instead, or a combo of `DirichletBC` (on some bdys) and `PorousFlowPiecewiseLinearSink` (on the other bdys), depending on your situation.  `DirichletBC` has the advantage of blindly agreeing with head observations, while the `Sink` has the advantage of agreeing with known fluxes.  If you know fluxes and/or heads then you can calibrate your hydraulic conductivity (permeability) to achieve the flows or heads observed.  When using the `sink` make sure your `flux_function` is chosen sensibly by checking the porepressure or head hasn't gone too crazy, and checking the fluxes out of the model are reasonable.

(2) Depending on your mesh and geometry, a `PorousFlowPolyLineSink` might be more appropriate, because the river could be thought of as a polyline.   Have a read about 1/3rd down this page for comments about perennial, ephemeral and rate-limited streams, and riverbed conductance: https://mooseframework.inl.gov/modules/porous_flow/sinks.html .  But it will accomplish essentially the same thing as your `PiecewiseLinearSink` (assuming your `boundary` for the `PiecewiseLinearSink` is set appropriately).  Also be aware that both these approaches can essentially fix porepressure or head at the nodal positions surrounding the `sink` if you choose a "strong" `flux_function`, but that could be exactly what you want, as the river can be thought of as an infinite supply or demand of water.

a

**Reply by josebastiase** [reply]
Hi Andy,

Thanks for your reply. I'm currently using a combination of  `PorousFlowPiecewiseLinearSink` and `DirichletBC` in the model:
`PorousFlowPiecewiseLinearSink`  for the top layer 
`DirichletBC` for all confined layers

The problem with `DirichletBC` is that the first layer of the model is unsaturated and drained, so a `DirichletBC` will fill with water the unsaturated zone and eventually water will leave the model from the top. Whereas with `PorousFlowPiecewiseLinearSink` I got more "control" so the latter doesn't happen, but is really hard to achieve a decent calibration.

**Comment by Traiwit** [comment]
Hi @josebastiase and @WilkAndy 

**not related to the q's, but on the same model**

Just wondering, how did you set the initial stress for this kind of problem?

I always have my geometry shrink due to gravity during the equilibrium step (1st step) using a simple rho*g*h.

```
[Kernels]
  [flux]
  type = PorousFlowAdvectiveFlux
  use_displaced_mesh = true
  variable = porepressure
  gravity = '0 0 -9.81'
  fluid_component = 0
[]
[]

  [./ini_xx]
    type = PiecewiseLinear
    axis = z
      x = '0 1000'
      y = '-29430000 0'
  [../]
  [./ini_yy]
    type = PiecewiseLinear
    axis = z
      x = '0 1000'
      y = '-29430000 0'
  [../]
  [./ini_zz]
    type = PiecewiseLinear
    axis = z
      x = '0 1000'
      y = '-19620000 0'  #2000*9.81*1000
  [../]

[./ini_stress]
    type = ComputeEigenstrainFromInitialStress
    eigenstrain_name = ini_stress
    initial_stress = 'ini_xx 0 0 0 ini_yy 0 0 0 ini_zz'
  [../]
```

Kind regards,
Traiwit

**Reply by WilkAndy** [reply]
Have a look at porous_flow/examples/coal_mining/coarse_with_fluid.i

a

**Reply by Traiwit** [reply]
Hi @WilkAndy,

I did, not sure where are these equations from, but I will give it a go, thanks!

```
  [./ini_pp]
    type = ParsedFunction
    vars = 'bulk p0 g    rho0'
    vals = '2E3 0.0 1E-5 1E3'
    value = '-bulk*log(exp(-p0/bulk)+g*rho0*z/bxculk)'
  [../]
  [./ini_xx]
    type = ParsedFunction
    vars = 'bulk p0 g    rho0 biot'
    vals = '2E3 0.0 1E-5 1E3  0.7'
    value = '0.8*(2500*10E-6*z+biot*(-bulk*log(exp(-p0/bulk)+g*rho0*z/bulk)))'
  [../]
  [./ini_zz]
    type = ParsedFunction
    vars = 'bulk p0 g    rho0 biot'
    vals = '2E3 0.0 1E-5 1E3  0.7'
    value = '2500*10E-6*z+biot*(-bulk*log(exp(-p0/bulk)+g*rho0*z/bulk))'
  [../]
```

**Reply by WilkAndy** [reply]
2500*10E-6*z is the gravitational load from a dry system.  biot is because you must prescribe *effective* stresses, not total stresses.  (-bulk * log ....) is a fancy way of doing hydrostatic: see https://mooseframework.inl.gov/modules/porous_flow/tests/gravity/gravity_tests.html
